pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -258,10 +258,9 @@ func (r *RootSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 	}
 	rs.Status.Reconciler = reconcilerRef.Name
 
-	// Get the latest deployment to check the status.
-	// For other operations, upsertDeployment will have returned the latest already.
-	if op == controllerutil.OperationResultNone {
-		deployObj, err = r.deployment(ctx, reconcilerRef)
+	// Get the latest deployment to check the status if current deployment pointer is nil.
+	if unstructuredDeployment == nil {
```

Comment:
Why would the return from upsertDeployment be nil?

Reply:
After we compare the the current deployment and declared deployment, if we find out no PATCH is needed, current logic will return nil and no error. We are converting the unstructured deployment we get at the beginning to delete several fields and convert back. I think it is better/ safer to just get a deployment directly for the status compute in this case.

Reply:
It's confusing (hard to read) to have a nil object returned mean there was no diff. That doesn't follow any of the existing client method conventions. What if you keep returning the operation too?

Reply:
Sure!